src/dune_rules/alias_rec.ml

```diff
@@ -1,4 +1,22 @@
 open Import
+module Alias_status = Action_builder.Alias_status
+module Lookup_alias = Action_builder.Lookup_alias
+
+module In_melange_target_dir = struct
+  let dep_on_alias_rec =
+    let rec map_reduce dir ~f =
```

Comment:
The name `map_reduce` is a bit weird here. It just looks like `reduce` to me without any mapping.